### ISSUE TITLE
Kernel Panic

### ISSUE THREAD

**Issue by explowz** [CachyOS/linux-cachyos#26]
Right after the update to kernel version 5.19.4, the server started experiencing frequent kernel panics.

```
systemd-shutdown[1]: Syncing filesystem and block devices - timed out, issuing SIGKILL to PID 2970.
systemd-shutdown[1]: Waiting for process 896 (systemd-journal), 2970 ((sd-sync))
systemd-shutdown[1]: Waiting for process 896 (systemd-journal), 2970 ((sd-sync))
INFO: task kworker/u171:1:157 blocked for more than 122 seconds.
          Not tainted 5.19.6-1-cachyos-cfs #1
"echo 0> /porc/sys/kernel/hung_task_timeout_secs" disables this message.
INFO: task kworker/u30:1:234 blocked for more than 122 seconds.
          Not tainted 5.19.6-1-cachyos-cfs #1
"echo 0> /porc/sys/kernel/hung_task_timeout_secs" disables this message.
INFO: task kworker/u30:1:650 blocked for more than 122 seconds.
          Not tainted 5.19.6-1-cachyos-cfs #1
"echo 0> /porc/sys/kernel/hung_task_timeout_secs" disables this message.
. . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . .
systemd-shutdown[1]: Remounting '/' timed out, issuing SIGKILL to PID 3036.
watchdog: watchdog0: watchdog did not stop!
Kernel panic - not syncing: 02: An NMI occurred. Depending on your system the reason for the NMI is logged in any one of the following resources:
1. Integrated Management Log (IML)
2. OA Syslog
3. OA Forward Progress Log
4. iLO Event Log
Kernel Offset: 0x9c00000 from oxffffffff81000000 (relocation range: oxffffffff81000000-oxffffffff81000000)
---[ end Kernel panic - not syncing: 02: An NMI occurred. Depending on your system the reason for the NMI is logged in any one of the following resources:
1. Integrated Management Log (IML)
2. OA Syslog
3. OA Forward Progress Log
4. iLO Event Log ]---
```

No, I do not have any of the logging resources listed above installed or configured.

I have tried using the default Arch Linux kernel, and that seems to have no problems.

My PKGBUILD configuration can be viewed on my fork [HERE](https://github.com/explowz/linux-cachyos/blob/master/linux-cachyos-cfs/PKGBUILD).

If anything else, like my modprobed database, etc., needs to be provided, I am able to upon request.

**Comment by sirlucjan** [comment]
If you can, install the kernel from our repository and see if the problem occurs.

**Comment by sirlucjan** [comment]
So check kernel from our repo and us let us know if it works.

**Comment by sirlucjan** [comment]
Please check our kernel first.

**Comment by sirlucjan** [comment]
See some more kernel from Arch - such as zen or regular. Then we will be sure if the problem lies with our kernel or if it should be looked for in the upstream.

**Comment by sirlucjan** [comment]
First, let's eliminate everything that is suspicious.

**Comment by sirlucjan** [comment]
> I forgot to mention, but the kernel has mitigations turned off (like @ptr1337 recommended), yet that didn't seem to cause any problems, since up until 5.19.4 everything worked perfectly.

This should have nothing to do with the error. We need to take a look along with @ptr1337 at what could be the problem.

**Comment by sirlucjan** [comment]
One last request - see also linux-zen. There is also a commit there that I suspect is causing the problems. If the problem does not occur, that too would be a clue to look for the problem elsewhere.

**Comment by sirlucjan** [comment]
Thanks for the tests. Will see with @ptr1337 what could be the problem.

**Comment by ptr1337** [comment]
That seems kinda odd to me. I will have a look with the upcoming release.
Thanks for the report. 

Which filesystem you are running ?

**Comment by ptr1337** [comment]
> I can firmly confirm that the same problems are persisting.

With the kernel from our repo or self compiled? Does gcc/llvm/lto does a difference ?

**Comment by ptr1337** [comment]
One thing - could you try to build a generic kernel?

So removing this line, so that it does not use the cpu optimatzion. I had on my intel buildserver something equal. Using "native-intel" in the config did solved this problem.

But weird that it worked before - but now not anymore.
Or test a kernel from our repo, you can simply download them with wget.
But you need a signing key from us, you can find in the README how this work

**Comment by ptr1337** [comment]
> The automatic march detection and changing the pacman.conf or manually?

Oh i missed the link. Sorry.

Setting both of them empty, so that just the generic is used:
https://github.com/CachyOS/linux-cachyos/blob/master/linux-cachyos-cfs/PKGBUILD#L93-L105

**Comment by ptr1337** [comment]
I maybe found a different issue which has been introduced with 5.19.5.

We have adapted from clearlinux a patch.
We did removed it now again for 5.19.7. May this introduced the issue.
https://github.com/CachyOS/kernel-patches/blob/master/5.19/0002-cachy.patch#L2201-L2215

**Comment by ptr1337** [comment]
I really dont get why this is happening. It seems really odd to me. Ive reverted some patches with 5.19.8 and added some fixes.
Have tested in on 4 devices - no regressions found so far on the tested devices.

May really download the Kernels from our repo as soon they are up, cause then I exactly know with which options they are built. Should be up in some hours

**Comment by ptr1337** [comment]
> > ![IMG_20220908_085514_587.jpg](https://user-images.githubusercontent.com/20655963/189045202-624627f4-29f4-4696-ac97-8e9ff1fee1cc.jpg)
> > This is what I got sent by the server keeper.
> 
> I have received some new information, that this actually happened when trying to boot into the default linux kernel provided by arch linux. This is the first time I've received such a report, and it seems to be due to the 5.19.7 update. They've tried loading it again, and this time I've been told that everything besides the on-screen cursor is frozen and does not work. This may've just turned into a bigger issue.

So, this is also happening with the default archlinux kernel? May report it to https://bugs.archlinux.org/.

Yes, you can checkout but you need to change the patch commit.
```
git checkout ef20e329bffc10745638dc50cf85e13b782ef5ba
cd linux-cachyos-cfs
wget https://raw.githubusercontent.com/CachyOS/kernel-patches/0a2e0a5cf03271ca5ef07ee58e7f1d3a38ecf296/5.19/all/0001-cachyos-base-all.patch
makepkg -sfi
```

If the checksums are failing, you can always also just do a `--skipchecksums`

Thanks for the testing, glad its not our fault.

**Comment by ptr1337** [comment]
Lets track this and see. May it got fixed with 5.19.8 but I did saw not a commit which should fix this

**Comment by ptr1337** [comment]
Glad to hear about! Have fun with the Kernel!
